reactioncommerce/reaction-admin
# Add layout for primary/detail view

Resolves #223 
Impact: **major**
Type: **feature**

## Issue

A new layout component is needed to satisfy UI requirements for a secondary sidebar with a detail view.

## Solution

- Create the Primary/Detail component
- Use new layout on Navigation dashboard
- Add helper hooks
- Refactor Dashboard
- Refactor route registration

## Screenshots

Mobile with the "more" button

![image](https://user-images.githubusercontent.com/42217/75739553-007bb080-5cba-11ea-80fb-57d1f19a7070.png)

Opened menu

![image](https://user-images.githubusercontent.com/42217/75739619-391b8a00-5cba-11ea-9efc-7ec794465f8f.png)

## Breaking changes

Non-breaking name changes when registering operator routes:
- Old => New
- `layoutComponent` => `LayoutComponent`
- `mainComponent` => `MainComponent`
- `isNavigationLink: true` => `group: "navigation"`
- `isSetting: true` => `group: "settings"`

Warnings will be logged when using the old properties. They'll automatically be transformed into their new values to retain compatibility.

## Testing
1. Run through the app and ensure the views load and look like they should with their layouts.
1. Pay specific attention to the navigation dashboard
1. Resize the nav dashboard add to see the "More" button appears
1. Click to open the menu

Keep in mind the nav editor drag-n-drop doesn't work in the Chrome mobile simulation. Any layout issues not related to the implementation of the Primary/Detail layout can go into another PR.
